File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {

    public static boolean isAlphaNum(char ch) {
        // return (Character.isAlphabetic(ch) || Character.isDigit(ch));
        return Character.isLetterOrDigit(ch);
    }

    public boolean isPalindrome(String s) {
        int i = 0, j = s.length() - 1;
        while (i < j) {
            char l, r;
            l = s.charAt(i);
            r = s.charAt(j);
            if (isAlphaNum(l) == false) {
                i++;
                continue;
            }
            if (isAlphaNum(r) == false) {
                j--;
                continue;
            }
            
            l = Character.toLowerCase(l);
            r = Character.toLowerCase(r);
            if (Character.compare(l, r) != 0) {
                return false;
            }
            i++;
            j--;
        }

        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:     
        if len(s.strip()) == 0:
            return True
        
        l, r = 0, len(s)-1        
        while l <= r:
            if not s[l].isalnum():
                while l < len(s) and not s[l].isalnum():
                    l += 1
            if not s[r].isalnum():
                while r >= 0 and not s[r].isalnum():
                    r -= 1                  
            if l > r:
                break
            if s[l].lower() != s[r].lower():
                return
            l += 1
            r -= 1
            
        return True
         